src/encoded/static/components/item-pages/ExperimentSetView.js

```diff
@@ -379,10 +380,39 @@ class QCMetricsTable extends React.PureComponent {
         super(props);
         this.memoized = {
             filterFilesWithQCSummary: memoize(commonFileUtil.filterFilesWithQCSummary),
-            groupFilesByQCSummaryTitles: memoize(commonFileUtil.groupFilesByQCSummaryTitles)
+            groupFilesByQCSummaryTitles: memoize(commonFileUtil.groupFilesByQCSummaryTitles),
+            filterQCSummaryItemsHavingTitleTooltips: memoize(this.filterQCSummaryItemsHavingTitleTooltips)
         };
     }
 
+    filterQCSummaryItemsHavingTitleTooltips(fileGroup) {
+        return _.filter(_.flatten(_.pluck(fileGroup, 'quality_metric_summary')), function (qmsItem) { return qmsItem.title_tooltip && qmsItem.title_tooltip.length > 0; });
+    }
+
+    renderForFileColValue(file) {
+        const fileAtId = file && object.atIdFromObject(file);
+        let fileTitleString;
+        if (file.accession) {
+            fileTitleString = file.accession;
+        }
+        if (!fileTitleString && fileAtId) {
+            var idParts = _.filter(fileAtId.split('/'));
+            if (idParts[1].slice(0, 5) === '4DNFI') {
+                fileTitleString = idParts[1];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=4901542)

Reply:
Opportunity for performance gain: This function doesn't depend on `this` so it could be a memoized static method.

Reply:
Well, nevermind, can't be memoized (at least on this component) since it called many times within component. But IMO should still be static if doesn't depend on "this" because makes comprehending this code much easier 1 year from now in future if look at it and say "whew, not a lot of instance methods/`this`-dependent-complexity" :-D .